client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/site-preview/index.tsx

```diff
@@ -1,19 +1,43 @@
+import { SiteThumbnail, Spinner } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
-import React, { FC } from 'react';
+import { FC, useEffect, useRef } from 'react';
 import { useQuery } from 'calypso/landing/stepper/hooks/use-query';
+import { useSitePreviewMShotImageHandler } from './hooks/use-site-preview-mshot-image-handler';
 import './style.scss';
 
 export const SitePreview: FC = () => {
 	const translate = useTranslate();
 	const fromUrl = useQuery().get( 'from' ) || '';
 
+	const previewRef = useRef< HTMLDivElement >( null );
+	const { mShotsOption, updateDimensions } = useSitePreviewMShotImageHandler();
+
+	useEffect( () => {
+		updateDimensions( previewRef );
+
+		// Throttle the resize event handling.
+		let resizeTimeout: ReturnType< typeof setTimeout >;
+		const throttledResizeHandler = () => {
+			clearTimeout( resizeTimeout );
+			resizeTimeout = setTimeout( () => updateDimensions( previewRef ), 200 );
+		};
+
+		window.addEventListener( 'resize', throttledResizeHandler );
+		return () => window.removeEventListener( 'resize', throttledResizeHandler );
+	}, [] );
+
 	return (
-		<div className="migration-instructions-from-preview">
-			<iframe
-				className="migration-instructions-from-preview__iframe"
-				title={ translate( 'Preview of the site being imported' ) }
-				src={ fromUrl }
-			/>
+		<div className="migration-instructions-from-preview" ref={ previewRef }>
+			<SiteThumbnail
+				mShotsUrl={ fromUrl }
+				className="migration-instructions-from-preview__screenshot"
```

Comment:
Since we're applying many styles as part of this class, would it be safer do add it in another div as wrapper of the `SiteThumbnail`?

So we avoid unexpected issues in case of updates in the `SiteThumbnail`. WDYT?

Reply:
Yap, that's also an option. As a fallback, I've updated the CSS here https://github.com/Automattic/wp-calypso/pull/92355/commits/9d25a04e71562611e54a5fa389f5f3f807e4196e in case the component changes and doesn't print the `classname` that we've passed. I've intentionally avoided having element tag in the selectors, rather used another class. Do you think this'll do?

Thanks for the suggestion!

Reply:
> in case the component changes and doesn't print the `classname` that we've passed

Ah, sorry! I meant something different than that. I meant trying to change the minimum possible of the styles of the `SiteThumbnail`, but testing now, I noticed it has a very opinative design and we really need to do override it, right?

That said, maybe we could just remove the new selector `& > .site-thumbnail`?